pypeit/coadd2d.py

```diff
@@ -1375,10 +1437,19 @@ def get_brightest_obj(self, specobjs_list, spat_ids):
                         objid_max[islit, iexp] = objid_this[imax]
                         bpm[islit, iexp] = False
 
+        # If a slit has bpm = True for some exposures and not for others, set bpm = True for all exposures
+        # get the slits that have bpm = True for any exposures
+        bmp_true_idx = np.array([np.any(b == True) for b in bpm])
+        if np.any(bmp_true_idx):
+            bpm_true = bpm[bmp_true_idx, :]
+            # If these slits have also bpm = False for some exposures, then bpm = True for all exposures
+            bpm_true[bpm_true == False] = True
+            bpm[bmp_true_idx, :] = bpm_true
```

Comment:
This is a little shorter, and I think equivalent (but untested!)...
```
# Find the rows where any of the bpm values are True
indx = np.any(bpm, axis=0)
if np.any(indx):
    # Flag all exposures in those rows
    bpm[indx,:] = np.ones((np.sum(indx),bpm.shape[1]), dtype=bool)
```

Reply:
Actually, for the last line, I think you can actually just do `bpm[indx,:] = True`.

Reply:
Thank you! Changed.